### PR TITLE
Add support for `replace_triggered_by` in `lifecycle`

### PR DESCRIPTION
**Pull request by lonegunmanb**
As #1388 said we need to support new feature in v1.2.0.

Hello @wata727 , this is the first time I submit pr to tflint repo, I'm not sure the whether this patch meets tflint's requirement, would you please give it a review since I have pending issue depends on `replace_triggered_by`? Thanks a lot!

### CODE REVIEW

**Review by bendrucker — APPROVED** [review]
The entire `terraform` package should be updated for 1.2, happy to merge that if you want to give it a shot. The intent of that package is to match TF and we won't accept individual patches.

**Review by bendrucker — CHANGES_REQUESTED** [review]
Accidentally set the previous review as approving

**Comment by wata727** [comment]
This individual patch is not accepted by itself, but we can start supporting v1.2 based on this patch. I will start working after merging this. Thanks.

**Pull request merged**
